Source/WebKit/Shared/AuxiliaryProcess.cpp

```diff
@@ -84,6 +84,11 @@ void AuxiliaryProcess::initialize(const AuxiliaryProcessInitializationParameters
 
     platformInitialize(parameters);
 
+    // Set process name before initializing the sandbox as we may need to
+    // connect to launch services to set the name and the sandbox doesn't
+    // necessarily allow it.
+    initializeProcessName(parameters);
```

Comment:
Perhaps you could create a sandbox extension for Launch Services, which would be revoked after setting the process name?